### PR TITLE
Layout component and initial conversion to pages router

### PR DESCRIPTION
**Pull request by 39otsu**
## Title
Layout component and initial conversion to pages router

## Issue Number
Touches on, possibly addressing, issue(s): #43 #61
Please request to meet to confirm if they are resolved. 

## Description
This pull request introduces a responsive layout component to our Next.js web application, ensuring that every page complies with responsive design principles. The `Layout` component has been enhanced to dynamically adjust its content based on the screen size, providing a seamless user experience across various devices.

With this change, introduces additional changes to the thought process in development. You can develop knowing that there is a clear structure. You can add page components and elements in a stacked on top of each other. The inner structure of those components can follow a localized ruleset. Perhaps, you want to stack them column-wise or grid, or even row-wise again.

Major Changes:
- Added logic to the `Layout` component to conditionally render header and footer based on props.
- Implemented responsive behavior in the `Header` component to display different layouts for screens wider and narrower than 1024px.
- Enhanced the `Footer` component to maintain visibility at the bottom of the page on all screen sizes.
- Changed homepage to use pages router. 

These changes aim to improve the overall usability and accessibility of our web application, especially for users accessing it on mobile devices and tablets.

The move to the page routers alters the convention used when development. Refer to the documentation for additional details on the pages router.
https://nextjs.org/docs/pages/building-your-application/routing

## Testing & Possible Errors
There are potential edge cases and interactions that might need further attention:
- The layout behavior on extremely small or large screens may need fine-tuning to ensure optimal presentation.
- Interaction between the layout components and other features of the application, such as routing and state management, should be monitored to address any unexpected issues.

## Images
BEFORE:
image

AFTER:

Desktop:
https://www.youtube.com/watch?v=DsGMoPav7rw

Mobile:
image

### CODE REVIEW

**Comment by AlvaroH2001** [comment]
When you get the chance can you do a git pull and then push so that the workflow checks do not fail based on test coverage.

**Review by AlvaroH2001 — APPROVED** [review]
This PR includes some crucial components to improve our UI and its functionality/responsiveness in different scenarios. Code looks good and from my visual inspection I can't find any faults. Has my approval.

**Review by aiharc1 — COMMENTED** [review]
Nice work reworking the layout modules. I am unsure why but the Vercel deployment does not run on my end. I am unable to verify that this implementation works at this time so am unable to approve right now. I'll check again to see if anything improves

**Comment by blake-gilmore** [comment]
Seems like the root URL no longer works. You have to add /login to the URL to get there now.

File: small-talk/components/header.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            
+
+            }
+        </div>
+    )
+}
+/*
+hover:bg-gradient-to-r from-cyan-500 to-blue-500
+*/
+
```
Comment (blake-gilmore) [inline small-talk/components/header.js:50]: Very cool. Love the conditional rendering. I learned something here!

**Review by blake-gilmore — APPROVED** [review]
LGTM, very excited to see this in action down the line.

**Pull request merged**